ethyca/fides
# Integrate fideslog for anonymous CLI analytics

Closes #336

### Code Changes

* [x] Populate IDs and default value for opt out
* [x] Build the opt out routine
* [x] Create fall back if `fides init` is not used
* [x] Include event tracking in the cli

### Steps to Confirm

* [x] `fidesctl init` generates the correct `fidesctl.toml`
* [x] A user is only asked one time if they would like to opt in
* [ ] Manually tested by at least one other person (add your name and check the box)

### Pre-Merge Checklist


### Description Of Changes

This PR is working slightly ahead of the release of fideslog, so some errors are expected until the package can be appropriately installed.

I believe this PR merging will coincide with the 1.0 release of fideslog


#### How to dive in:

##### Test the opt-out routine
* delete the `fidesctl.toml` file and run `fidesctl init`
    * This should recreate `fidesctl.toml` with the appropriate ID and an opt out value based on input
* delete the newly added values from `fidesctl.toml` (cli.analytics_id, user.analytics_opt_out)
* Run `fidesctl apply demo_resources/ --dry` (or any `fidesctl` command minus `init`)
    * The opt-out routine should be triggered and the appropriate values written to `fidesctl.toml`

##### Validating event data
* Once opted in, any cli events should be written to Snowflake (see information in 1pass)

Catches:
* `init` only command not included with event tracking
